AirCastingTests/SearchAndFollowTests/SearchViewModelTests.swift

```diff
@@ -0,0 +1,56 @@
+// Created by Lunar on 03/08/2022.
+//
+import XCTest
+@testable import AirCasting
+
+class SearchViewModelTests: XCTestCase {
+    var sut: SearchViewModel!
+    
+    override func setUp() {
+        super.setUp()
+        sut = .init()
+    }
+    
+    func test_onInitialization_onlyParticulateMatterShouldBeSelected() {
+        let selected = sut.measurementTypes.filter({ $0.isSelected })
+        XCTAssertEqual(selected.count, 1)
+        XCTAssertEqual(selected.first?.name, "Particulate Matter")
+    }
+    
+    func test_whenSensorTapHappens_shouldSelectTapped() {
+        sut.sensorTypes = [.init(isSelected: false, name: "1"),
+                           .init(isSelected: true, name: "2"),
+                           .init(isSelected: false, name: "3")]
+        
+        let tappingSensorName = "1"
+        
+        sut.onSensorTap(with: tappingSensorName)
+        
+        let selected = sut.sensorTypes.filter({ $0.isSelected })
+        
+        XCTAssertEqual(selected.count, 1)
+        XCTAssertEqual(selected.first!.name, tappingSensorName)
```

Comment:
I'm not sure if tests go forward after one assertion fails, but if yes, then the test will crash if selected.count is 0

Reply:
I tested and as I can see using breakpoints, it goes forward but the fail message which I got was connected to the line that ensures there is one element inside. I am going to investigate a bit and make myself sure about its correctness

Reply:
Ania is right, this code will crash, failures don't stop execution. You can use `XCTUnwrap` or just do optional unwrap in possible:
```Swift
XCTAssertEqual(selected.filrst?.name, tappingSensorName)
```